pkg/apis/v1beta1/nodepool.go

```diff
@@ -78,6 +79,40 @@ type Disruption struct {
 	// +kubebuilder:validation:Schemaless
 	// +optional
 	ExpireAfter NillableDuration `json:"expireAfter"`
+	// Budgets is a list of Budgets.
+	// If there are multiple active budgets, Karpenter uses
+	// the most restrictive maxUnavailable. If left undefined,
+	// this will default to one budget with a maxUnavailable to 10%.
+	// +kubebuilder:validation:XValidation:message="'crontab' must be set with 'duration'",rule="!self.all(x, (has(x.crontab) && !has(x.duration)) || (!has(x.crontab) && has(x.duration)))"
+	// +kubebuilder:default:={{maxUnavailable: "10%"}}
+	// +kubebuilder:validation:MaxItems=50
+	// +optional
+	Budgets []Budget `json:"budgets,omitempty" hash:"ignore"`
+}
+
+// Budget defines when Karpenter will restrict the
+// number of Node Claims that can be terminating simultaneously.
+type Budget struct {
+	// MaxUnavailable dictates how many NodeClaims owned by this NodePool
+	// can be terminating at once. It must be set.
+	// This only considers NodeClaims with the karpenter.sh/disruption taint.
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:default:="10%"
+	MaxUnavailable intstr.IntOrString `json:"maxUnavailable" hash:"ignore"`
```

Comment:
didn't we discuss "value" for this?

Reply:
The consensus was to do MaxUnavailable. It matches really nicely with the fact that the parent struct name is budgets, relying on the established parallel with PDB -> MaxUnavailable

Reply:
MaxUnavailable only make sense in comparison to MinAvailable.

Reply:
Also -- MaxUnavailable is a bit confusing to me, since it implies what we know what the maximum value is.